### PR TITLE
GTextAlignmentCenter

### PR DESCRIPTION
**Pull request by CTKRocks**
Centers by adding the `glyph->advance`s up to get the width of the specified text, and taking the `box.size.w / 2` and subtracting `text_size.w / 2` to set the `line_origin`.

### CODE REVIEW

File: lib/neographics/src/text/text.c [hunk 1]
```diff
@@ -220,21 +227,37 @@ void n_graphics_draw_text(
...
             continue;
         }
         n_GGlyphInfo *glyph = n_graphics_font_get_glyph_info(font, text[i]);
-        width += glyph->width;
-        height += glyph->height;
+        width += glyph->advance;
         
         i++;
     }
+    printf("TEXT: %s \nWIDTH: %d \n", debug_text, width);
```
Comment (jwise) [inline lib/neographics/src/text/text.c]: remove printf?

File: lib/neographics/src/text/text.h [hunk 2]
```diff
@@ -44,7 +44,7 @@ typedef enum n_GTextOverflowMode {
 
 typedef enum n_GTextAlignment {
     n_GTextAlignmentLeft = 0,
-    n_GTextAlignmentCenter,
+    n_GTextAlignmentCenter = 1,
```
Comment (jwise) [inline lib/neographics/src/text/text.h]: I think this should not be necessary

File: lib/neographics/src/text/text.c [hunk 3]
```diff
@@ -220,21 +227,37 @@ void n_graphics_draw_text(
 
 n_GSize n_graphics_text_layout_get_content_size(const char * text, n_GFont const font)
 {
-    double width = 0;
-    double height = 0;
-    int i = 0;
-    while (text[i] != '\0')
+    uint32_t text_len = 0;
+    while (text[text_len] != '\0')
```
Comment (jwise) [inline lib/neographics/src/text/text.c]: `strlen(text)`

File: lib/neographics/src/text/text.c [hunk 4]
```diff
@@ -220,21 +227,37 @@ void n_graphics_draw_text(
...
+n_GSize n_graphics_text_layout_get_content_size_with_index(const char *text, n_GFont const font, uint32_t idx, uint32_t idx_end)
+{
+    uint8_t width = 0;
+    uint8_t height = 0;
+    uint32_t i = idx;
+    
+    char debug_text[100];
+    int debug_i = 0;
+    
+    while (i <= idx_end)
```
Comment (jwise) [inline lib/neographics/src/text/text.c]: `for (i = idx; i <= idx_end; i++)`

File: rwatch/ui/window.c [hunk 5]
```diff
@@ -19,7 +19,7 @@ Window *window_create()
     Window *window = calloc(1, sizeof(Window));
     if (window == NULL)
     {
-        SYS_LOG("window", APP_LOG_LEVEL_ERROR, "No memory for Window");
```
Comment (jwise) [inline rwatch/ui/window.c]: were we tripping on this?

File: rwatch/ui/layer/menu_layer.c [hunk 6]
```diff
@@ -433,7 +432,7 @@ void menu_cell_chalk_draw(GContext *ctx, const Layer *layer, const char *previou
     {
         GFont font = fonts_get_system_font(FONT_KEY_GOTHIC_18_BOLD);
         n_GSize text_size = n_graphics_text_layout_get_content_size(previous, font);
-        GRect subtitle_rect = GRect((frame.size.w / 2) - (text_size.w / 2) - 10, 20, frame.size.w, 24);
+        GRect subtitle_rect = GRect(0, 20, frame.size.w, 24);
```
Comment (jwise) [inline rwatch/ui/layer/menu_layer.c]: 💯

File: lib/neographics/src/text/text.c [hunk 7]
```diff
@@ -220,21 +227,37 @@ void n_graphics_draw_text(
...
     {
+        debug_text[debug_i] = text[i];
+        debug_i++;
```
Comment (jwise) [inline lib/neographics/src/text/text.c]: debug_text still useful?

File: lib/neographics/src/text/text.c [hunk 8]
```diff
@@ -98,20 +98,19 @@ void n_graphics_draw_text(
...
         // 0b110xxxxx 0b10xxxxxx
         // 0b1110xxxx 0b10xxxxxx 0b10xxxxxx
         // 0b11110xxx 0b10xxxxxx 0b10xxxxxx 0b10xxxxxx
-        if (text[index] == '\n'
-                && (char_origin.x + (__CODEPOINT_NEEDS_HYPHEN_AFTER(codepoint) ? hyphen->advance : 0)
-                        <= box.origin.x + box.size.w)) {
-            n_graphics_prv_draw_text_line(ctx, text,
-                line_begin, index, font, line_origin);
+        
+        if (text[index] == '\n' && (char_origin.x + (__CODEPOINT_NEEDS_HYPHEN_AFTER(codepoint) ? hyphen->advance : 0) <= box.origin.x + box.size.w)) {
```
Comment (jwise) [inline lib/neographics/src/text/text.c]: any particular reason why these lines were unbroken?

File: lib/neographics/src/common.c [hunk 9]
```diff
@@ -31,7 +31,10 @@ void n_graphics_set_pixel(n_GContext * ctx, n_GPoint p, n_GColor color) {
         &ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x / 8],
         p.x % 8, (color.argb & 0b111111));
 #else
-    ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x] = color.argb;
+    if (p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x >= 0)
```
Comment (jwise) [inline lib/neographics/src/common.c]: hmm ... any chance we could take this off the hot path?  (do bounding in the outer loop?)

File: lib/neographics/src/common.c [hunk 10]
```diff
@@ -31,10 +31,7 @@ void n_graphics_set_pixel(n_GContext * ctx, n_GPoint p, n_GColor color) {
         &ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x / 8],
         p.x % 8, (color.argb & 0b111111));
 #else
-    if (p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x >= 0)
-    {
-        ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x] = color.argb;
-    }
+    ctx->fbuf[p.y * __SCREEN_FRAMEBUFFER_ROW_BYTE_AMOUNT + p.x] = color.argb;
```
Comment (jwise) [inline lib/neographics/src/common.c]: :100: Can you rebase and squash these changes?

File: lib/neographics/src/text/text.c (hunk 8 above)
Comment (CTKRocks) [reply lib/neographics/src/text/text.c]: No, I personally like it on one line and my editor was indenting me weird. I can revert

**Pull request merged**